backend/apps/slack/events/member_joined_channel/gsoc.py

```diff
@@ -1,75 +1,52 @@
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-def gsoc_handler(event: dict, client: WebClient, ack) -> None:
-    """Slack #gsoc new member handler.
-
-    Args:
-        event (dict): The event payload from Slack.
-        client (slack_sdk.WebClient): The Slack WebClient instance.
-        ack (Callable): The acknowledgment function to confirm event processing.
+class Gsoc(EventBase):
+    """Slack GSoC channel join event handler."""
 
-    """
-    from apps.slack.common.gsoc import GSOC_GENERAL_INFORMATION_BLOCKS
+    event_type = "member_joined_channel"
 
-    ack()
-    if not settings.SLACK_EVENTS_ENABLED:
-        return
+    def __init__(self):
+        """Initialize the GSoC event handler."""
+        super().__init__()
+        self.matchers = [lambda event: f"#{event['channel']}" == OWASP_GSOC_CHANNEL_ID]
 
-    user_id = event["user"]
+    def handle_event(self, event, client):
+        """Handle the member_joined_channel event for the GSoC channel."""
+        user_id = event["user"]
 
-    try:
-        conversation = client.conversations_open(users=user_id)
-    except SlackApiError as e:
-        if e.response["error"] == "cannot_dm_bot":
-            logger.warning("Error opening conversation with bot user %s", user_id)
-            return
-        raise
+        client.chat_postEphemeral(
+            blocks=GSOC_2025_MILESTONES,
+            channel=event["channel"],
+            user=user_id,
+            text=get_text(GSOC_2025_MILESTONES),
+        )
 
-    client.chat_postEphemeral(
-        blocks=GSOC_2025_MILESTONES,
-        channel=event["channel"],
-        user=user_id,
-    )
+        conv = self.open_conversation(client, user_id)
+        if conv:
+            context = {
+                "user_id": user_id,
+                "gsoc_channel_id": OWASP_GSOC_CHANNEL_ID,
+                "gsoc_info_blocks": GSOC_GENERAL_INFORMATION_BLOCKS,
+                "FEEDBACK_CHANNEL_MESSAGE": FEEDBACK_CHANNEL_MESSAGE,
+            }
 
-    blocks = (
-        markdown(
-            f"Hello <@{user_id}> and welcome to <{OWASP_GSOC_CHANNEL_ID}> channel!{NL}"
-            "Here's how you can start your journey toward contributing to OWASP projects and "
-            "making the most of Google Summer of Code:"
-        ),
-        *GSOC_GENERAL_INFORMATION_BLOCKS,
-        markdown(
-            "🎉 We're excited to have you on board, and we can't wait to see the amazing "
-            "contributions you'll make! Happy contributing and good luck with your GSoC "
-            "journey!"
-        ),
-        markdown(f"{FEEDBACK_CHANNEL_MESSAGE}"),
-    )
-    client.chat_postMessage(
-        blocks=blocks,
-        channel=conversation["channel"]["id"],
-        text=get_text(blocks),
-    )
+            client.chat_postMessage(
+                blocks=self.get_render_blocks(context),
+                channel=conv["channel"]["id"],
+                text=get_text(self.get_render_blocks(context)),
+            )
```

Comment:
This seems to be a common logic that could be added into the base class. Check the commands implementation approach and use the same method structure for context/template and general content generation approach.